### PR TITLE
replace vue compiler for vue 3 compatibility

### PR DESCRIPTION
**Pull request by lzurbriggen**
I replaced the old `vue-template-compiler` with the new vue 3 `@vue/compiler-sfc` to prevent version mismatch errors when using this package with vue 3.

(also cleaned up eslint errors)

I'm not sure how you want to handle dependencies/releases. `compiler-sfc` is on `3.0.0-rc.5`, if you accept the change maybe this should be a `next` version until there's a stable `3.0.0` release.

### CODE REVIEW

**Review by vperron — APPROVED** [review]
Agreed ! With 3 small gotchas though.

- I think you're gonna have to rebase this as soon as #92  is merged :/ sorry ! but it should be quick.
- it would be probably way better to separate the style (eslint) updates in a different commit
- @vue/compiler-sfc now has a definitive v3.0.0 :)

**Comment by vperron** [comment]
There you go @lzurbriggen :) A few small more steps :+1:

**Comment by lzurbriggen** [comment]
@vperron I updated the compiler dependency and split code style changes to a second commit (indentation adjustments in the tests are mostly because of changes in the compiler output).

File: src/javascript-extract.js
Comment (vperron) [inline src/javascript-extract.js:30]: I don't understand this change, can you maybe place a comment to explain why it's so ?

File: src/extract.js
Comment (vperron) [inline src/extract.js:571]: I don't understand why this whole renaming of `content` to `filecontent` is necessary, especially if you rename it to `content` at the very end ^^ 

Either finish the job and rename everything to `fileContent` in a separate commit (paying attention to the camelCase convention that exists in this package) or don't do it ?

**Comment by vperron** [comment]
That is fantastic ! I paid attention to do a second review, and seen a few little gotchas. ONce you address them, I'll gladly merge this immediately :partying_face:

File: src/extract.js
Comment (lzurbriggen) [reply src/extract.js:571]: I honestly didn't remember (it's been a while). The original code violates the `no-param-reassign` (and `no-shadow` on line 276) eslint rule.

I'll rename it to fileContent.

File: src/javascript-extract.js
Comment (lzurbriggen) [reply src/javascript-extract.js:30]: I should have paid closer attention when rebasing. I tried to fix a problem (`guard-for-in` eslint violation) that has been solved in the `added babel parser support for javascript parsing` PR. I'll remove it.

File: src/extract.js
Comment (vperron) [reply src/extract.js]: Oh, that's interesting.

**Comment by vperron** [comment]
Thanks for your contributions @lzurbriggen  :) I made you a write-approved contributor to this repository !

**Pull request merged**
